cmd/kube-apiserver/app/options/options.go

```diff
@@ -68,7 +69,7 @@ type ServerRunOptions struct {
 func NewServerRunOptions() *ServerRunOptions {
 	s := ServerRunOptions{
 		GenericServerRunOptions: genericoptions.NewServerRunOptions(),
-		Etcd:                 genericoptions.NewEtcdOptions(kubeoptions.DefaultEtcdPathPrefix, api.Scheme, nil),
+		Etcd:                 genericoptions.NewEtcdOptions(storagebackend.NewDefaultConfig(kubeoptions.DefaultEtcdPathPrefix, api.Scheme, nil)),
```

Comment:
I don't like that. We construct the storagebackend config behind the scenes and use it from `EtcdOptions` to create `RESTOptionsFactories`. How do you want to customize the backend? Patching the config after creating the EtcdOptions is dirty, especially if you pass prefix, scheme and codec before. Using NewDefaultConfig is using clean composition. What's the issue with that?